### ISSUE TITLE
[argo-rollouts] Update CRDs to helm 3

### ISSUE THREAD

**Issue by mubarak-j** [argoproj/argo-helm#1164]
### Describe the bug

My understanding is that Helm v3 doesn't support upgrading or deleting CRDs in helm releases, therefore current settings like `installCRDs` and `keepCRDs` no longer needed or apply. And for the same reason, CRDs should not be templated similar to other argo charts in this repo. The same CRDs probably need to be sync'd with upstream manifests as well. 

### Related helm chart

argo-rollouts

### To Reproduce

argo-rollouts helm chart hasn't fully migrated to helm v3, mostly around CRDs.

### Expected behavior

consistent CRDs install method with helm v3 and other charts in this repo.

**Comment by mkilchhofer** [comment]
Hmm the rollouts chart follows a different pattern to manage CRDs. This also works with helm v3 IMHO. All other charts (argo-cd, argo-events, argo-workflows) uses the default and well-known pattern where the CRDs live in the directory `/crds`. This way helm does not update the CRDs if they changed. In `argo-cd` this is required as the chart supports deploying an instance of these CRDs (`server.additionalApplications` and `server.additionalProjects`).

Do you know about this document? https://github.com/helm/community/blob/main/hips/hip-0011.md

**Comment by mubarak-j** [comment]
@mkilchhofer Yes, that where i built my current understanding of helm3 and how it manages CRDs. Argo-Rollout also deploy manages CR of type `analysis-run` and `experiments`. As per the document you have provided, helm3 doesn't support templating in CRDs or manage their life-cycle aside for first install. and therefore current settings like:
```yaml
{{- if .Values.keepCRDs }}
"helm.sh/resource-policy": keep
```
are no longer valid. Please take a look at this document: https://helm.sh/docs/chart_best_practices/custom_resource_definitions/

> This is also the reason behind removing templating support from CRDs.

Im one version behind on rollout chart version but when i run helm diff or sync, the update to CRDs dont show or apply because helm3 completely ignores CRDs that already installed, which is what expected in helm3 per that hdocs

**Comment by mkilchhofer** [comment]
> Argo-Rollout also deploy manages CR of type `analysis-run` and `experiments`.
Yes, but theses CRs (an instance of the CRDs) are not deployed via the chart. They are coming from users via API or separate charts (or kustomize).

> As per the document you have provided, helm3 doesn't support templating in CRDs or manage their life-cycle aside for first install. and therefore current settings like:
> 
> ```yaml
> {{- if .Values.keepCRDs }}
> "helm.sh/resource-policy": keep
> ```
> 
> are no longer valid. Please take a look at this document: https://helm.sh/docs/chart_best_practices/custom_resource_definitions/

For sure helm supports it. Helm supports templating all kind of Yamls as long as they live in the `/templates` directory. The annotation `"helm.sh/resource-policy": keep` is generic for all resource types and is still supported: https://helm.sh/docs/howto/charts_tips_and_tricks/#tell-helm-not-to-uninstall-a-resource.
Only the CRD related annotation is gone in Helm v3 (`"helm.sh/hook": crd-install`).
 
> Im one version behind on rollout chart version but when i run helm diff or sync, the update to CRDs dont show or apply because helm3 completely ignores CRDs that already installed, which is what expected in helm3 per that hdocs

Hmmm helm-diff is a plugin right? I think then the plugin does not support CRDs completely.


The helm chart for rook-ceph also uses this: https://github.com/rook/rook/blob/master/deploy/charts/rook-ceph/templates/resources.yaml#L7
This annotation were added in PR https://github.com/argoproj/argo-helm/pull/922 by @Temikus "recently" (about 6 month) ago. And they work as intended IMHO, even with helm v3.

**Comment by mubarak-j** [comment]
Thank you @mkilchhofer for clarification
